src/components/profile-view/profile-view.jsx

```diff
@@ -0,0 +1,169 @@
+        }
+      })
+      .catch(error => {
+        console.error("Error updating profile:", error);
+        alert("Error updating profile. Please try again later.");
+      });
+  };
+
+  const handleDeregister = () => {
+    if (
+      !window.confirm(
+        "Are you sure you want to delete your account? This action cannot be undone."
+      )
+    ) {
+      return;
+    }
+
+    fetch(
+      `https://my-movie-app-ab91e4bb4611.herokuapp.com/users/${user.username}`,
+      {
+        method: "DELETE",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${token}`,
+        },
+      }
+    )
+      .then(response => {
+        if (response.ok) {
+          alert("Your account has been deleted.");
+          localStorage.removeItem("user");
+          localStorage.removeItem("token");
+          window.location.href = "/"; // Redirect to signup page or any other page
+        } else {
+          alert("Failed to delete your account. Please try again later.");
+        }
+      })
+      .catch(error => {
+        console.error("Error deleting account:", error);
+        alert("Error deleting account. Please try again later.");
+      });
+  };
+
+  return (
+    <div>
+      <h1>{user.username}'s Profile</h1>
+      <Form onSubmit={handleSubmit}>
+        <Form.Group controlId="formUsername">
+          <Form.Label>Username:</Form.Label>
+          <Form.Control
+            type="text"
+            value={updatedUsername}
+            onChange={e => setUpdatedUsername(e.target.value)}
+          />
+        </Form.Group>
+        <Form.Group controlId="formPassword">
+          <Form.Label>Password:</Form.Label>
+          <Form.Control
+            type="password"
+            value={updatedPassword}
+            onChange={e => setUpdatedPassword(e.target.value)}
+          />
+        </Form.Group>
+        <Form.Group controlId="formEmail">
+          <Form.Label>Email:</Form.Label>
+          <Form.Control
+            type="email"
+            value={updatedEmail}
+            onChange={e => setUpdatedEmail(e.target.value)}
+          />
+        </Form.Group>
+        <Form.Group controlId="formBirthday">
+          <Form.Label>Birthday:</Form.Label>
+          <Form.Control
+            type="date"
+            value={updatedBirthday}
+            onChange={e => setUpdatedBirthday(e.target.value)}
+          />
+        </Form.Group>
+        <Button variant="primary" type="submit">
+          Update Profile
+        </Button>
+        <Button variant="danger" onClick={handleDeregister}>
+          Deregister
+        </Button>
+      </Form>
+      <h2>Favorite Movies</h2>
+      <div className="favorite-movies">
+        {favoriteMovies.length === 0 ? (
+          <p>No favorite movies added yet.</p>
+        ) : (
+          <ul>
+            {favoriteMovies.map(movie => (
+              <li key={movie.id}>{movie.title}</li>
+            ))}
+          </ul>
+        )}
```

Comment:
Consider reusing the movie card component here...

<img width="1728" alt="Screenshot 2024-07-26 at 11 22 30 PM" src="https://github.com/user-attachments/assets/afbb5bbb-cff2-4583-b9a6-5249107239d4">